src/DataValues/DecimalValue.php

```diff
@@ -353,6 +353,24 @@ public static function newFromArray( $data ) {
 	}
 
 	/**
+	 * Create DecimalValue from regular numeric string or value.
+	 * @param int|float|string $number
+	 *
+	 * @return self
+	 */
+	public static function makeDecimalValue( $number ) {
+
+		if ( is_string( $number ) && $number !== '' ) {
+			if ( $number[0] !== '-' && $number[0] !== '+' ) {
+				$number = '+' . $number;
+			}
+		}
+
+		return new static( $number );
+	}
```

Comment:
The constructor already accepts ints and floats, therefore I do not see why it should not accept a simple string like "1234" with no leading plus sign. Please move to the constructor and add tests.

The nested `if`s are a bit weird in my opinion. Should be after the `trim` and more like `if ( ctype_digit( $number[0] ) )`.